sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexAsyncClient.java

```diff
@@ -0,0 +1,831 @@
+                                                                            Context context) {
+        AutocompleteRequest autocompleteRequest = createAutoCompleteRequest(
+            searchText, suggesterName, autocompleteOptions);
+        return new PagedFluxBase<>(
+            () -> this.autocompleteFirst(requestOptions, autocompleteRequest, context));
+    }
+
+    private Mono<AutocompletePagedResponse> autocompleteFirst(RequestOptions requestOptions,
+                                                              AutocompleteRequest autocompleteRequest,
+                                                              Context context) {
+        return restClient.documents()
+            .autocompletePostWithRestResponseAsync(autocompleteRequest, requestOptions, context)
+            .map(AutocompletePagedResponse::new);
+    }
+
+    /**
+     * Retrieve the first page of a document search
+     *
+     * @param searchRequest the search request
+     * @param requestOptions the request options
+     * @param context the context to associate with this operation.
+     * @return {@link Mono}{@code <}{@link PagedResponse}{@code <}{@link SearchResult}{@code >}{@code >} next page
+     * response with results
+     */
+    private Mono<SearchPagedResponse> searchFirstPage(SearchRequest searchRequest,
+                                                      RequestOptions requestOptions,
+                                                      Context context) {
+        return restClient.documents()
+            .searchPostWithRestResponseAsync(searchRequest, requestOptions, context)
+            .map(SearchPagedResponse::new);
+    }
+
+    /**
+     * Retrieve the next page of a document search
+     *
+     * @param searchRequest the search request
+     * @param nextPageParameters json string holding the parameters required to get the next page:
+     *                           skip is the number of documents to skip, top is the number of documents per page.
+     *                           Due to a limitation in PageFlux, this value is stored as String and converted
+     *                           to its Integer value
+     *                           before making the next request
+     * @param context the context to associate with this operation.
+     * @return {@link Mono}{@code <}{@link PagedResponse}{@code <}{@link SearchResult}{@code >}{@code >} next page
+     * response with results
+     */
+    private Mono<SearchPagedResponse> searchNextPage(
+        SearchRequest searchRequest,
+        RequestOptions requestOptions,
+        String nextPageParameters,
+        Context context) {
+
+        if (CoreUtils.isNullOrEmpty(nextPageParameters)) {
+            return Mono.empty();
+        }
+
+        // Extract the value of top and skip from @search.nextPageParameters in SearchPagedResponse
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
We need to rethink the implementation of this method, since it's duplicating paging logic that's already in the service, and that logic may change.

The contract is that the client should use exactly the payload returned in `@search.nextPageParameters` as the request body in the POST request for the next page of search results. If we want to avoid this client breaking if the service paging behavior changes, we should respect that contract.

That said, the risk of these details changing is low in the short term. As long as this implementation works, I would be ok shipping at least a preview with it, but we should address this sooner rather than later.